File: code_samples/typescript/testing_unit_tests.ts
/**
 * Testing: Unit Tests
 * AI/ML Training Sample
 */

interface IUnitTests {
    data: string;
    process(input: string): void;
    validate(): boolean;
}

class UnitTests implements IUnitTests {
    data: string;
    
    constructor() {
        this.data = "";
    }
    
    process(input: string): void {
        this.data = input;
    }
    
    getData(): string {
        return this.data;
    }
    
    validate(): boolean {
        return this.data.length > 0;
    }
}

// Example usage
const instance = new UnitTests();
instance.process("example");
console.log(`Data: ${instance.getData()}`);
console.log(`Valid: ${instance.validate()}`);

export { UnitTests, IUnitTests };

File: code_samples/elixir/testing_unit_tests.ex
# Testing: Unit Tests
# AI/ML Training Sample

defmodule UnitTests do
  defstruct data: ""
  
  def new(), do: %UnitTests{}
  
  def process(%UnitTests{} = struct, input) do
    %{struct | data: input}
  end
  
  def get_data(%UnitTests{data: data}), do: data
  
  def validate(%UnitTests{data: data}) do
    String.length(data) > 0
  end
end

# Example usage
instance = UnitTests.new()
updated = UnitTests.process(instance, "example")
IO.puts("Data: " <> UnitTests.get_data(updated))
IO.puts("Valid: " <> to_string(UnitTests.validate(updated)))
